dev/source-release.sh

```diff
@@ -20,72 +20,137 @@
 
 set -e
 
-if [ -z "$1" ]; then
-  echo "Usage: $0 <version-num> <rc-num>"
-  exit
-fi
-
-if [ -z "$2" ]; then
-  echo "Usage: $0 <version-num> <rc-num>"
-  exit
+usage () {
+    echo "usage: $0 [-k <key_id>] [-r <git_remote>] <version_num> <rc_num>"
+    echo "  -k      Specify signing key. Defaults to \"GPG default key\""
+    echo "  -r      Specify Git remote name. Defaults to \"origin\""
+    echo "  -d      Turn on DEBUG output"
+    exit 1
+}
+
+# Default repository remote name
+remote="origin"
+
+while getopts "k:r:d" opt; do
+  case "${opt}" in
+    k)
+      keyid="${OPTARG}"
+      ;;
+    r)
+      remote="${OPTARG}"
+      ;;
+    d)
+      set -x
+      ;;
+    *)
+      usage
+      ;;
+  esac
+done
+
+shift $((OPTIND-1))
+
+version="$1"
+rc="$2"
```

Comment:
Should we just make these part of the `getops` workflow or would deviating too much from the current structure be undesirable?

Reply:
Done